docs/source/developers/guide/resources.rst

```diff
@@ -27,3 +27,50 @@
 ************************************
 Additional information and resources
 ************************************
+
+On this page we will list resources interesting for a contributor
+that is not yet acquainted with different parts of Apache Arrow.
+
+Additional information
+----------------------
+
+- `CMake <https://cmake.org/documentation/>`_
+- `Gandiva <https://arrow.apache.org/blog/2018/12/05/gandiva-donation/>`_
+- SIMD
+
+  - `Columnar is Fast section of Apache Arrow Overview <https://arrow.apache.org/overview/>`_
+  - `SIMD on Wikipedia <https://en.wikipedia.org/wiki/SIMD>`_
+
+- Memory management
+
+  - `C++ <https://arrow.apache.org/docs/cpp/memory.html>`_
+  - `Python <https://arrow.apache.org/docs/python/memory.html>`_
+  - `jemalloc <https://arrow.apache.org/blog/2018/07/20/jemalloc/>`_,
+    `mimalloc <https://github.com/microsoft/mimalloc#readme>`_
+  - `Plasma <https://arrow.apache.org/blog/2017/08/08/plasma-in-memory-object-store/>`_
+
+- `Snappy <https://github.com/google/snappy#introduction>`_
```

Comment:
I'm not sure I understand the selection which is done here. Arrow has many optional dependencies (mimalloc, Snappy...) and it's not obvious that giving links to only a couple of them is useful.

Reply:
The selection was done completely from my side due to the fact I bumped into this group when building PyArrow. I could try to list all of them (will go through the cpp part of the dev docs).

Reply:
I created two JIRA issues [ARROW-15130](https://issues.apache.org/jira/browse/ARROW-15130) and [ARROW-15129](https://issues.apache.org/jira/browse/ARROW-15129) to remove random information listed here. I will link to the Glossary once it is done.